src/Orleans/IDs/TypeCodeMapper.cs

```diff
@@ -7,10 +7,10 @@ namespace Orleans.Runtime
     /// </summary>
     internal static class TypeCodeMapper
     {
-        internal static GrainClassData GetImplementation(Type interfaceType, string grainClassNamePrefix = null)
+        internal static GrainClassData GetImplementation(Type interfaceType, IRuntimeClient runtimeClient, string grainClassNamePrefix = null)
         {
             GrainClassData implementation;
-            IGrainTypeResolver grainTypeResolver = RuntimeClient.Current.GrainTypeResolver;
+            IGrainTypeResolver grainTypeResolver = runtimeClient.GrainTypeResolver;
```

Comment:
Fixed, kind-of. I can pass `runtimeClient.GrainTypeResolver` directly, but I cannot inject `IGrainTypeResolver` directly into `GrainFactory`, since on clients, `IGrainTypeResolver` is created during initialization (it involves a remote call to a silo).

I agree that `IRuntimeClient` is a bit too broad in scope - we should pare it down.